### PR TITLE
Cmdct 4528 error fix

### PR DESCRIPTION
**Pull request by jessabean**
### Description
Fixes a bug where marking all analysis methods as "not utilized" caused the analysis methods page to crash

### Related ticket(s)

CMDCT-4528

---
### How to test
Already created a report in deployed env: 

1. In Section 1 > Analysis methods (/naaar/state-and-program-information/analysis-methods), edit an analysis method.
2. For the question "Is this analysis method utilized to assess plan compliance?" answer **Yes** and fill out the rest of the fields in the drawer.
3. Save and close the drawer.
4. Edit the remaining analysis methods and answer **No**.
5. Edit the analysis method marked as **Yes** and change the answer to **No**.
6. Save and close the drawer.
7. The page should still render, you should see no errors, and the table row should say "Not utilized"

**Note:** this is a workaround to an edge case that still needs to be solved for. That user flow is pending product requirements.

### Notes

---
### Pre-review checklist

- [x] I have added [thorough](https://shorturl.at/aejkF) tests, if necessary
- [ ] I have updated relevant documentation, if necessary
- [x] I have performed a self-review of my code
- [x] I have manually tested this PR in the deployed cloud environment

---
#### Review
- [ ] Design: This work has been reviewed and approved by design, if necessary
- [ ] Product: This work has been reviewed and approved by product owner, if necessary

#### Security
_If either of the following are true, notify the team's ISSO (Information System Security Officer)._

- [ ] These changes are significant enough to require an update to the SIA.
- [ ] These changes are significant enough to require a penetration test.
---

_convert to a different template: [test → val](?expand=1&template=test-to-val-deployment.md)_ | _[val → prod](?expand=1&template=val-to-prod-deployment.md)_

### CODE REVIEW

File: services/ui-src/src/utils/testing/mockRouter.tsx [hunk 1]
```diff
@@ -2,5 +2,7 @@ import React from "react";
 import { BrowserRouter as Router } from "react-router-dom";
 
 export const RouterWrappedComponent: React.FC = ({ children }) => (
-  <Router>{children}</Router>
+  <Router future={{ v7_startTransition: true, v7_relativeSplatPath: true }}>
```
Comment (jessabean) [inline services/ui-src/src/utils/testing/mockRouter.tsx:5]: Went ahead and fixed up console warnings because they were driving me crazy. See: https://reactrouter.com/upgrading/v6

File: services/ui-src/src/components/reports/DrawerReportEntityRows.test.tsx [hunk 2]
```diff
@@ -59,83 +69,95 @@ describe("<DrawerReportEntityRow />", () => {
...
-  describe("Analysis methods custom logic", () => {
-    const mockAnalysisMethodEntityStore: McrEntityState = {
-      entities: [],
-      entityType: EntityType.ANALYSIS_METHODS,
-      selectedEntity: {
-        id: "k9t7YoOeTOAXX3s7qF6XfN33",
-        name: "Geomapping",
-        isRequired: true,
+  describe("NAAAR Analysis Methods", () => {
+    const incompleteAnalysisMethodsReport = {
```
[the remark this answers is not in the document]
Comment (gmrabian) [reply services/ui-src/src/components/reports/DrawerReportEntityRows.test.tsx:73]: If they are reused elsewhere then yes. Otherwise probably fine to just have here

**Review by gmrabian — APPROVED** [review]
does the fix

**Review by bangbay-bluetiger — APPROVED** [review]
